src/api/providers/fetchers/modelCache.ts

```diff
@@ -44,15 +44,25 @@ async function readModels(router: RouterName): Promise<ModelRecord | undefined>
  */
 export const getModels = async (options: GetModelsOptions): Promise<ModelRecord> => {
 	const { provider } = options
-	let models = memoryCache.get<ModelRecord>(provider)
+
+	// kilocode_change start: cacheKey
+	const cacheKey = JSON.stringify(options)
```

Comment:
Isn't this one of those huge all-encompassing objects?  Are we really going to invalidate the cache for every unrelated options change?

Brainstorm aka probably harder than it looks: can't we "just" respect the HTTP caching rules, and let some http client deal with caching?

Reply:
just relying on HTTP caching sounds great tbh, the main difficulty would be selling it upstream I think